src/main/java/org/jabref/gui/collab/ExternalChangesResolverViewModel.java

```diff
@@ -100,8 +102,15 @@ public void acceptMergedChange(DatabaseChange databaseChange) {
     }
 
     public void applyChanges() {
-        changes.stream().filter(DatabaseChange::isAccepted).forEach(change -> change.applyChange(ce));
-        ce.end();
-        undoManager.addEdit(ce);
+        if (changes.stream().noneMatch(change -> !change.isAccepted())) {
+            changes.stream().filter(DatabaseChange::isAccepted).forEach(change -> change.applyChange(ce));
+            ce.end();
+            undoManager.addEdit(ce);
+            stateManager.activeTabProperty().get().get().updateTabTitle(false);
```

Comment:
does not work in all cases, because dialog ist also called from BackupUIManager

![image](https://github.com/JabRef/jabref/assets/1366654/98253828-f967-478b-b27e-1b3c10a6bd81)

Reply:
I could not see where the asterisk is really written.

Thus, I propose:

DatabaseChangeMonitor, line 62:

Extract `new DatabaseChangesResolverDialog` to own variable `databaseChangesResolverDialog`.
After `showCustomDialogAndWait`, ask the `databaseChangesResolverDialog` whether there werre changes etc. -- Modify the `setResultConverter` to return something more meaningful than `true` and `false`. maybe a record with `allChangesResolved: boolean` and `allExternalChangesAccepted: boolean`. These can then be used for the subsequent checks and steps.


I am not sure about: Remove the undoManager thing from `viewModel.applyChanges(); (because I think, it makes no sense in a backuip.

Reply:
can you please give some clarity on what cases it will not work?
from what I understand even if BackupUIManager calls it this should work right?
also I didn't get the point of implementing checks after `showCustomDialogAndWait `. is it to stop the file monitoring if changes are resolved and accepted?

Reply:
OK, I tried. Deleted entry "Corti_2009". Waited until backup is written. Closed JabRef (no saving). Deleted `.bak-discared` file. Started JabRef. Denied "Deleted entry Corti_2009". Selected Edit/undo. Removed the asterisk, but does not remove the entry itself. Which is kind of OK. (The reason is `BibDatabaseContext originalDatabase = originalParserResult.getDatabaseContext();`) - undo/redo has to be done at `showReviewBackupDialog`. - Future work needs to work on this - and pass the changes from `DatabaseChangesResolverDialog` back so that they can be ported to the library -- or we just accept that undo/redo does not work here ^^

---

The issue is different: In the case of a backup, the dialog displays changes introduced by the backup. If all (!) changes are rejected, there is no change in the file (and no asterisk has to be added). Reason: it is the BACKUP file, not the file on disk. Thus, the logic is different than from the file on disk.

Idea to solve this: Pass the changes from `ExternalChangesResolverViewModel` up to `DatabaseChangesResolverDialog`, where the `EasyBind.subscribe` in initialize stores it into a local variable. (`applyChanges` is removed from `ExternalChangesResolverViewModel`). The `resultConverter` (of `DatabaseChangesResolverDialog`) returns the changes. The caller in `DatabaseChangeMonitor` and `BackupUIManager` can handle appropriately. The code of `applyChanges` from `ExternalChangesResolverViewModel` moves there. I know, that 3 lines of code will be duplicated, but I accept hat. In the case of the Backupmanager, there is no need to `resetChangedProperties()` in the case of no changes - the library is not changed directly after loading.